ts/hooks/useScreenEndMargin.tsx

```diff
@@ -0,0 +1,57 @@
+import { IOSpacingScale, IOVisualCostants } from "@pagopa/io-app-design-system";
+import { useMemo } from "react";
+import { useSafeAreaInsets } from "react-native-safe-area-context";
+
+type EndScreenSpacingValues = {
+  screenEndSafeArea: number;
+  screenEndMargin: number;
+};
+
+/**
+ * A custom React Hook that returns two spacing values that must be applied at
+ * the end of a ScrollView to prevent the content from being cut off.
+ * Depending on what you need, you can set one of these two values as `paddingBottom`
+ * using ScrollView's `contentContainerStyle` prop:
+ * - `screenEndSafeArea`
+ *    The amount of safe area without additional margins. For devices that don't have safe area
+ *    boundaries (e.g. iPhone with home button) it returns a fallback value that prevents content
+ *    from sticking to the bottom.
+ * - `screenEndMargin`
+ *    The total amount of space to add at the end of the ScrollView. It's a sum of the
+ *    `screenSafeArea' value and the default `contentEndMargin' that should be applied
+ *    at the end of each app screen.
+ */
+export const useScreenEndMargin = (): EndScreenSpacingValues => {
+  const insets = useSafeAreaInsets();
+
+  const needSafeAreaMargin = useMemo(() => insets.bottom !== 0, [insets]);
```

Comment:
As we discussed live, the use of `useMemo` here does not improve the code execution, since it is caching the result of an operation that is not expensive at all

Reply:
@Vangaorth Applied in the following commit → https://github.com/pagopa/io-app/pull/5808/commits/1805be83ab0cba63454008855478668f996b8ec6
Since I've already used similar patterns in `FooterActions` and `IOScrollView`, I've changed them as well:
- **`FooterActions`** → https://github.com/pagopa/io-app/pull/5808/commits/b0ef3cf69ebe8a3a89060a7eb0dcb1a63bfbd9a0
- **`IOScrollView`** → https://github.com/pagopa/io-app/pull/5808/commits/8500df000b45e4487257094a0c256cc2a9f002c8